src/container/include/container/container_parsing.hpp

```diff
@@ -0,0 +1,107 @@
+#ifndef CLIPPER_CONTAINER_PARSING_HPP
+#define CLIPPER_CONTAINER_PARSING_HPP
+
+#include <cstring>
+#include <vector>
+
+#include <container/datatypes.hpp>
+
+namespace clipper {
+
+namespace container {
+
+template <typename T>
+struct input_parser_type {
+  static const bool is_string_parser = false;
+};
+
+template <>
+struct input_parser_type<char> {
+  static const bool is_string_parser = true;
+};
+
+template <typename D>
```

Comment:
What is this template parameter D? Can you add a comment that documents what the meaning of the template type is?